scripts/etc/init.d/rcS

```diff
@@ -528,7 +528,13 @@ echo "OK"
 # default hostname
 echo -n "  Configuring hostname... "
 echo -n $hostname > /rootfs/etc/hostname || fail
-echo "127.0.0.1	localhost" > /rootfs/etc/hosts || fail
+if [ ! -f /rootfs/etc/hosts ]; then
+    echo "127.0.0.1	localhost" > /rootfs/etc/hosts || fail
+else
+    if ! grep -q "localhost" /rootfs/etc/hosts; then
+        echo "127.0.0.1	localhost" >> /rootfs/etc/hosts
+    fi
```

Comment:
even if `localhost` is set to something different than `127.0.0.1` :thought_balloon: